internal/provider-config/config.go

```diff
@@ -442,6 +444,20 @@ func (p *LxdProviderConfig) acceptRemoteCertificate(remoteName string, url strin
 		return err
 	}
 
+	if token != "" {
+		// Decode token.
+		rawToken, err := shared.CertificateTokenDecode(token)
+		if err != nil {
+			return fmt.Errorf("Failed decoding trust token for remote %q: %v", remoteName, err)
+		}
+
+		// Compare token and certificate fingerprints.
+		certFingerprint := shared.CertFingerprint(cert)
+		if certFingerprint != rawToken.Fingerprint {
```

Comment:
That feels like something the LXD client should be doing itself, no?

Doing the token decoding + verification in the provider feels like a layer violation to me. I guess my intention was to simply error out if a token was provided along with asking to blindly accept the remote cert. In pseudo-code that'd be:

```
if token != "" && p.acceptServerCertificate {
    return errors.New("Using a remote add token is incompatible with accept_remote_certificate=true".)
}
```

Reply:
Hmm, but then the user would need to have a server certificate already present locally?

The current approach makes the `acceptRemoveCertificate` flag redundant when using tokens, as the remote certificate fingerprint will be always compared to the one within the token. Therefore, whenever the fingerprint matches, the certificate is accepted.

Reply:
After discussing this at length, you are right, peaking at the token and doing the fingerprint comparison needs to be done by the provider because the LXD Go API doesn't provide a more straightforward way of doing this.